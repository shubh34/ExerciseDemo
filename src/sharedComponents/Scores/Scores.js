import './Scores.scss';
import React from 'react';
import PropTypes from 'prop-types';


const Scores = (props) => {
	const { score, votes, className } = props;
	return (
		<div className="score">
			<span className={`score_rate ${className}`}>
				<span className="icon__rate">{score}</span>
			</span>
			<span className={`score_counter ${className}`}>
				<span className="icon__profile">{votes}</span>
			</span>
		</div>
	);
};

Scores.propTypes = {
	score: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
	votes: PropTypes.string.isRequired,
	className: PropTypes.string,
};
export default Scores;
