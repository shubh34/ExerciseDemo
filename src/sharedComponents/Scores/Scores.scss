.score {
    text-align: center;
    padding: 8px
}

.icon__rate ::before {
content: '\e61a';
}

.icon__profile ::before {
content: '\e60e';
}
.score_rate, .score_counter {
    padding: 4px;
    font-size: 14px;
    letter-spacing: 2px;
}
.score_rate { 
    color: #ffe300;
}
.movie-details-score {
    font-size: 20px;
}