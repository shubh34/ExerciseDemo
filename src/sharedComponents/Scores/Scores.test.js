import React from 'react';
import { shallow } from 'enzyme';
import Scores from './Scores';

it('renders without crashing', () => {
	const div = shallow(<Scores score='2' votes="2000" />);
	expect(div).toMatchSnapshot();
});
