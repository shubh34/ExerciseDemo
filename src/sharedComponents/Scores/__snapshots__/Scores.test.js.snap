// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="score"
>
  <span
    className="score_rate undefined"
  >
    <span
      className="icon__rate"
    >
      2
    </span>
  </span>
  <span
    className="score_counter undefined"
  >
    <span
      className="icon__profile"
    >
      2000
    </span>
  </span>
</div>
`;
